### PR TITLE
fix(restart): supply a random node as provider when restarting a seed

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -2429,26 +2429,6 @@ def disrupt_memory_stress(self):
...
-
-        else:
-            super(NotSpotNemesis, self).set_target_node()
-
-        self.log.info('Current Target: %s', self.target_node)
-
-    def disrupt(self):
-        raise NotImplementedError()
-
-
```
Comment (roydahan) [inline sdcm/nemesis.py:2451]: First, this change is not correct and not requested.
The idea with this class is to not run this nemesis when the instances are Spot because we cannot stop spot instance.

Second, even if you think this change is required, the commit message should have been much clearer and explanatory on why the change is required.

**Review by roydahan — CHANGES_REQUESTED** [review]
Remove the second commit from this PR.

**Comment by roydahan** [comment]
Has the "replace seed" change been tested in a longevity?
Is it working? (meaning 7588 can be solved like that).

File: sdcm/nemesis.py (hunk 1 above)
Comment (ShlomiBalalis) [reply sdcm/nemesis.py:2451]: When I tried to run the longevity only with RestartThenRepairNodeMonkey, it failed due to this class. When I asked @fruch about it, he said that this nemesis is unnecessary and that I should remove it in a second commit

Comment (fruch) [reply sdcm/nemesis.py:2451]: Yes this `NotSpotNemesis` should go, but it's not the goal of this PR, it's a bit backwards the goal is addressing scylladb/scylla#7588

**Comment by ShlomiBalalis** [comment]
> Has the "replace seed" change been tested in a longevity?
> Is it working? (meaning 7588 can be solved like that).

@roydahan Yes, I ran a longevity for it and it ran 15 times. There was only a small issue with c-s at the end of the run, but I don't think it's related

File: sdcm/nemesis.py (hunk 1 above)
Comment (roydahan) [reply sdcm/nemesis.py:2451]: > When I tried to run the longevity only with RestartThenRepairNodeMonkey, it failed due to this class. When I asked @fruch about it, he said that this nemesis is unnecessary and that I should remove it in a second commit

Maybe you tried with a spot instance?

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -2577,7 +2557,7 @@ def disrupt(self):
         self.disrupt_stop_start_scylla_server()
 
 
-class RestartThenRepairNodeMonkey(NotSpotNemesis):
+class RestartThenRepairNodeMonkey(Nemesis):
```
Comment (roydahan) [inline sdcm/nemesis.py:2560]: Change that back as well, or better remove the unnecessary commit.

File: sdcm/cluster_aws.py [hunk 3]
```diff
@@ -643,6 +644,22 @@ def restart(self):
...
+
+            seed_provider = [{
+                "class_name": "org.apache.cassandra.locator.SimpleSeedProvider",
+                "parameters": [{
+                    "seeds": f"{random_node.ip_address}"
+                }, ],
+            }, ]
+
+            with self.remote_scylla_yaml() as scylla_yml:
+                scylla_yml["seed_provider"] = seed_provider
```
Comment (roydahan) [inline sdcm/cluster_aws.py:662]: Need the same for other clusters types, not only aws.

**Review by roydahan — CHANGES_REQUESTED** [review]
Remove unrelated commit + implement for other backend types or for base cluster.

File: sdcm/nemesis.py (hunk 2 above)
Comment (fruch) [reply sdcm/nemesis.py:2560]: @roydahan I approved this removal, this code path is only relevant when this specific nemesis class is used in all other case is not being used.

People run into it only when class is used, and it's not up to date in all the logic needed.

It's only add confusion.

File: sdcm/cluster_aws.py (hunk 3 above)
Comment (fruch) [reply sdcm/cluster_aws.py:662]: Right now other backend doesn't support this nemesis, i.e. the replace flow is only supported by AWS backend
Moving this only the seed logic into shared place, won't be helpful at this stage.

**Review by fruch — APPROVED** [review]
LGTM,
Other backends would need more work to have this flow supported, fort example GCE doesn't have yet the code for in place replace_address.
And for k8s it's not relevant, since operator is handling such things (as far as I know)

File: sdcm/nemesis.py (hunk 1 above)
Comment (ShlomiBalalis) [reply sdcm/nemesis.py:2451]: Nope, on demand
